vms/platformvm/utxo/camino_handler_test.go

```diff
@@ -106,3 +109,421 @@ func TestVerifySpendUTXOsWithLocked(t *testing.T) {
+			ins: []*avax.TransferableInput{
+				{
+					Asset: avax.Asset{ID: h.ctx.AVAXAssetID},
+					In: &secp256k1fx.TransferInput{
+						Amt: 1,
+						Input: secp256k1fx.Input{
+							SigIndices: []uint32{0},
+						},
+					},
+				},
+			},
+			outs: []*avax.TransferableOutput{
+				{
+					Asset: avax.Asset{ID: h.ctx.AVAXAssetID},
+					Out: &secp256k1fx.TransferOutput{
+						Amt: 1,
+					},
+				},
+			},
+			creds: []verify.Verifiable{
+				&secp256k1fx.Credential{
+					Sigs: [][crypto.SECP256K1RSigLen]byte{
+						credentials[0].Sigs[0],
+					},
+				},
+			},
+			signers: []verify.State{
+				&secp256k1fx.TransferOutput{
+					Amt: 1,
+					OutputOwners: secp256k1fx.OutputOwners{
+						Addrs: []ids.ShortID{
+							address0,
+						}, Threshold: 1,
+					},
+				},
+			},
+			producedAmounts: map[ids.ID]uint64{},
+			expectedErr:     nil,
+		},
+		"One Signer, Threshold=2": {
+			utxos: []*avax.UTXO{
+				{
+					Asset: avax.Asset{ID: h.ctx.AVAXAssetID},
+					Out: &secp256k1fx.TransferOutput{
+						Amt: 1,
+					},
+				},
+			},
+			ins: []*avax.TransferableInput{
+				{
+					Asset: avax.Asset{ID: h.ctx.AVAXAssetID},
+					In: &secp256k1fx.TransferInput{
+						Amt: 1,
+						Input: secp256k1fx.Input{
+							SigIndices: []uint32{0, 1},
+						},
+					},
+				},
+			},
+			outs: []*avax.TransferableOutput{
+				{
+					Asset: avax.Asset{ID: h.ctx.AVAXAssetID},
+					Out: &secp256k1fx.TransferOutput{
+						Amt: 1,
+					},
+				},
+			},
+			creds: []verify.Verifiable{
+				&secp256k1fx.Credential{
+					Sigs: [][crypto.SECP256K1RSigLen]byte{
+						credentials[1].Sigs[0],
+					},
+				},
+			},
+			signers: []verify.State{
+				&secp256k1fx.TransferOutput{
+					Amt: 1,
+					OutputOwners: secp256k1fx.OutputOwners{
+						Addrs: []ids.ShortID{
+							address1,
+							address2,
+						}, Threshold: 1,
+					},
+				},
+			},
+			producedAmounts: map[ids.ID]uint64{},
+			expectedErr:     secp256k1fx.ErrTooManySigners,
+		},
+		"Two Signers, Threshold=2": {
+			utxos: []*avax.UTXO{
+				{
+					Asset: avax.Asset{ID: h.ctx.AVAXAssetID},
+					Out: &secp256k1fx.TransferOutput{
+						Amt: 1,
+					},
+				},
+			},
+			ins: []*avax.TransferableInput{
+				{
+					Asset: avax.Asset{ID: h.ctx.AVAXAssetID},
+					In: &secp256k1fx.TransferInput{
+						Amt: 1,
+						Input: secp256k1fx.Input{
+							SigIndices: []uint32{0, 1},
+						},
+					},
+				},
+			},
+			outs: []*avax.TransferableOutput{},
+			creds: []verify.Verifiable{
+				&secp256k1fx.Credential{
+					Sigs: [][crypto.SECP256K1RSigLen]byte{
+						credentials[1].Sigs[0],
+						credentials[2].Sigs[0],
+					},
+				},
+			},
+			signers: []verify.State{
+				&secp256k1fx.TransferOutput{
+					Amt: 1,
+					OutputOwners: secp256k1fx.OutputOwners{
+						Addrs: []ids.ShortID{
+							address1,
+							address2,
+						}, Threshold: 2,
+					},
+				},
+			},
+			producedAmounts: map[ids.ID]uint64{},
+			expectedErr:     nil,
+		},
+		"Three Signers, Threshold=2": {
```

Comment:
Not really relevant to Msig.
I'd like to propose a change to make it such.

Add all 3 addresses as owners in signer (again important they are sorted `add1 < add2 < add3` There is handy method in `ids` https://github.com/chain4travel/caminogo/blob/a9cdd869bbcd15d267bfa86fd2364256789d22f5/ids/short.go#L114
- `Threshold=2`
- Creds: `2 out of 3 signatures and correct sigIndices`
Then we can  repeat a test same as 👆 but with different set of Creds, so let say this test `key1 & key2` signed,
then in following tests: `key2 & key3`, `key1 & key3`. So then tests will demonstrate the MSig is working as expected.